src/state/translations-mixin.ts

```diff
@@ -339,13 +339,16 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
           ...data,
         },
       };
-      const changes: Partial<HomeAssistant> = {
-        resources,
-        localize: await computeLocalize(this, language, resources),
-      };
+
+      // Update resources immediately, so when a new update comes in we don't miss values
+      this._updateHass({ resources });
+
+      const localize = await computeLocalize(this, language, resources);
 
       if (language === (this.hass ?? this._pendingHass).language) {
-        this._updateHass(changes);
+        this._updateHass({
```

Comment:
This if is not needed, we return on L333 if the language does not match

Reply:
But there is an await in between